files/query_cache.py

```
@@ -0,0 +1,423 @@
+        int: New version number
+    """
+    try:
+        version_key = _get_cache_version_key(scope, identifier)
+
+        # Try atomic increment first (Redis, Memcached)
+        try:
+            new_version = cache.incr(version_key)
+            # Set TTL on the key if this is first incr
+            cache.expire(version_key, CACHE_VERSION_TIMEOUT)
+            logger.debug(f"Bumped cache version for {scope}:{identifier} to {new_version}")
+            return new_version
```

Ruff (0.14.4)

Consider moving this statement to an `else` block

(TRY300)